File: Logica_de_Programacao/PalindromoRecursivo.java
public class PalindromoRecursivo {
    public static boolean isFim(String s) {
        return (s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
    }

    public static boolean palindromo(String texto, int i) {
        Boolean palindromo = true;

        // condição de parada
        if (i >= ((texto.length() - 1) / 2)) {

            if (texto.charAt(i) != texto.charAt(texto.length() - 1 - i)) {
                palindromo = false;
            } else {
                // chamada recursiva
                palindromo(texto, i - 1);
            }

        }
        return palindromo;
    }

    public static void main(String[] args) {
        Boolean palindromo = true;
        String texto = MyIO.readLine();

        while (!isFim(texto)) {
            MyIO.println(palindromo(texto, texto.length() - 1) ? "SIM" : "NAO");
            texto = MyIO.readLine();
        }
    }
}

File: Logica_de_Programacao/PalindromoRecursivo.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#define boolean short
#define true 1
#define false 0

boolean isFim(char s[])
{
    return (s[0] == 'F' && s[1] == 'I' && s[2] == 'M');
}

boolean palindromo(char texto[], int i, int j)
{

    // condição de parada
    if (i >= ((strlen(texto) - 1) / 2))
    {
        if (texto[i] != texto[j])
        {
            return false;
        }
        else
        {
            // chamada recursiva
            palindromo(texto, i - 1, j + i);
        }
    }
    return true;
}

int main()
{
    char texto[1000];
    fgets(texto, 1000, stdin);

    while (!isFim(texto))
    {
        printf(palindromo(texto, strlen(texto) - 2, 0) ? "SIM\n" : "NAO\n");
        fgets(texto, 1000, stdin);
    }
}
